src/progress/progress_set.rs

```diff
@@ -193,6 +194,29 @@ impl ProgressSet {
         self.next_configuration = None;
     }
 
+    pub(crate) fn restore_conf_states(
+        &mut self,
+        conf_states: &[ConfStateWithIndex],
+        next_idx: u64,
+        max_inflight: usize,
+    ) {
+        if conf_states.is_empty() {
+            return;
+        }
+        let mut meta = SnapshotMetadata::default();
+        let len = conf_states.len();
+        if conf_states[len - 1].in_membership_change {
```

Comment:
This is not verbose. why not just `meta.set_conf_state(conf_states[0])`. And set next if there is additional values.